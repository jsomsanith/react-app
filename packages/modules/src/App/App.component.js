import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

export default function App({ loading, RootComponent, store }) {
	return (
		<Provider store={store}>
			<Suspense fallback={loading}>
				<RootComponent />
			</Suspense>
		</Provider>
	);
}
App.propTypes = {
	loading: PropTypes.node,
	RootComponent: PropTypes.func,
	store: PropTypes.object,
};
