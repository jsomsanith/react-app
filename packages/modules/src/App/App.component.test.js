import React from 'react';
import { shallow } from 'enzyme';
import App from './App.component';

describe('App component', () => {
	it('should render', () => {
		// when
		const wrapper = shallow(
			<App
				loading={<div id="loading">Please wait</div>}
				RootComponent={() => <div id="root">Content</div>}
			/>,
		);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
