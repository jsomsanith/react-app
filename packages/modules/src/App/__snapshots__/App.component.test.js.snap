// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
<Provider>
  <UNDEFINED
    fallback={
      <div
        id="loading"
      >
        Please wait
      </div>
    }
  >
    <RootComponent />
  </UNDEFINED>
</Provider>
`;
